docs: Clarify how to switch to Next auth middleware (#1162)
t3-oss/create-t3-app: www/src/pages/en/usage/next-auth.md

## Usage with Next.js middleware

Usage of NextAuth.js with Next.js middleware [requires the use of the JWT session strategy](https://next-auth.js.org/configuration/nextjs#caveats) for authentication. This is because the middleware is only able to access the session cookie if it is a JWT. By default, Create T3 App is configured to use the **default** database strategy, in combination with Prisma as the database adapter.  

> After switching to the JWT session strategy. Make sure to update the `session` callback in your `[...nextauth].ts`.  

Reviewer: It should not be `src/server/auth.ts` ?
In new version of the create-t3-app, all the auth configuration stuff is outside of the `[...nextauth].ts`

Author: Sure! Wow, I started using T3 app just a week ago. 😅 
```suggestion
After switching to the JWT session strategy. Make sure to update the `session` callback in your `src/server/auth.ts`.  
```

The `user` object will be `undefined`. Instead, retrieve the user's ID from the `token` object.  
I.e.:

> ```diff:pages/api/[...nextauth].ts

Author: ```suggestion
```diff:server/auth.ts```

## Usage with Next.js middleware

Usage of NextAuth.js with Next.js middleware [requires the use of the JWT session strategy](https://next-auth.js.org/configuration/nextjs#caveats) for authentication. This is because the middleware is only able to access the session cookie if it is a JWT. By default, Create T3 App is configured to use the **default** database strategy, in combination with Prisma as the database adapter.  
After switching to the JWT session strategy. Make sure to update the `session` callback in `src/server/auth.ts`.  
The `user` object will be `undefined`. Instead, retrieve the user's ID from the `token` object.  
I.e.:

> `diff:server/auth.ts`

Reviewer 2: ```suggestion
```diff:server/auth.ts
```

> ...
> export const authOptions: NextAuthOptions = {
> 
> - session: {
> - strategy: "jwt",
> - },
>   callbacks: {
> 
> * session({ token, user }) {
> 
> - session({ token, session }) {
> 
> *     if (session.user) {
> *       session.user.id = user.id;
> 
> -     if (session.user && token.sub) {
> -       session.user.id = token.sub;
>         }
>         return session;
>       }
>   },
>   };
> 

Reviewer 2: ```suggestion
  export const authOptions: NextAuthOptions = {
+   session: {
+     strategy: "jwt",
+   },
    callbacks: {
-     session({ session, user }) {
+     session({ session, token }) {
-       if (session.user) {
+       if (session.user && token.sub) {
-         session.user.id = user.id;
+         session.user.id = token.sub;
        }
        return session;
      }
    },
  };
```

| NextAuth.js Docs                  | https://next-auth.js.org/               |
| NextAuth.js GitHub                | https://github.com/nextauthjs/next-auth |
| tRPC Kitchen Sink - with NextAuth | https://kitchen-sink.trpc.io/next-auth  |

> ```

Reviewer 2: ```suggestion
```
